src/ngraph/op/fused/batch_to_space.cpp

```diff
@@ -0,0 +1,208 @@
+    vector<int64_t> upperbounds_values;
+    auto flat_node_shape = flat_node->get_shape();
+    for (size_t i = 0; i < flat_node_shape.size(); ++i)
+    {
+        upperbounds_values.push_back(flat_node_shape.at(i) - crops_end_values.at(i));
+    }
+    const auto upperbounds = op::Constant::create(
+        crops_end.get_element_type(), Shape{upperbounds_values.size()}, upperbounds_values);
+
+    vector<int64_t> begin_mask(data_shape.size(), 0);
+    vector<int64_t> end_mask(data_shape.size(), 0);
+    flat_node = make_shared<op::v1::StridedSlice>(
+        flat_node, crops_begin_const, upperbounds, begin_mask, end_mask);
+    return NodeVector{flat_node};
+}
+
+void ngraph::op::v1::BatchToSpace::pre_validate_and_infer_types()
+{
+    PartialShape data_pshape = get_input_partial_shape(0);
+
+    auto data = input_value(0);
+    auto block = input_value(1);
+    auto crops_begin = input_value(2);
+    auto crops_end = input_value(3);
+    NGRAPH_CHECK(block.get_node_shared_ptr()->is_constant(),
+                 "block_shape input node is expected to be a static constant");
+
+    NGRAPH_CHECK(crops_begin.get_node_shared_ptr()->is_constant(),
+                 "crops_begin input node is expected to be a static constant");
+
+    NGRAPH_CHECK(crops_end.get_node_shared_ptr()->is_constant(),
+                 "crops_end input node is expected to be a static constant");
+
+    const auto& data_type = get_input_element_type(0);
+    const auto& block_shape_type = get_input_element_type(1);
+    const auto& crops_begin_type = get_input_element_type(2);
+    const auto& crops_end_type = get_input_element_type(3);
+    NODE_VALIDATION_CHECK(this,
+                          block_shape_type == element::i32 || block_shape_type == element::i64,
+                          "block_shape element type must be either int64_t or int32_t but got (",
+                          block_shape_type,
+                          ").");
+
+    NODE_VALIDATION_CHECK(this,
+                          crops_begin_type == element::i32 || crops_begin_type == element::i64,
+                          "crops_begin element type must be either int64_t or int32_t but got (",
+                          crops_begin_type,
+                          ").");
+
+    NODE_VALIDATION_CHECK(this,
+                          crops_end_type == element::i32 || crops_end_type == element::i64,
+                          "crops_end element type must be either int64_t or int32_t but got (",
+                          crops_end_type,
+                          ").");
+
+    if (data_pshape.is_dynamic())
+    {
+        set_output_type(0, data_type, PartialShape::dynamic());
+    }
+}
+
+std::shared_ptr<ngraph::Node>
+    ngraph::op::v1::BatchToSpace::copy_with_new_args(const ngraph::NodeVector& new_args) const
+{
+    if (new_args.size() != 4)
+    {
+        throw ngraph_error("Incorrect number of new arguments");
+    }
```

Comment:
Could you use this helper here?
`check_new_args_count(this, new_args);`

Reply:
fixed